Makefile

```diff
@@ -1,8 +1,20 @@
+bin_dir = bin
+build_dir = .build
 package = purescript
 exe_target = purs
 stack_yaml = STACK_YAML="stack.yaml"
 stack = $(stack_yaml) stack
 
+.DEFAULT_GOAL := help
+
+$(bin_dir)/hlint: ci/install-hlint.sh
+	BIN_DIR=$(bin_dir) BUILD_DIR=$(build_dir) $<
+	touch $@
+
+clean: ## Remove build artifacts
```

Comment:
Is this intended to cover all of the places that a build creates files? I know there are many others, most of which I had to clean while switching the polykinds PR on and off: `.test_modules`, `**/.stack-work`, `tests/support/node_modules`, `tests/support/bower_components`, etc. A `git clean -fdX` takes care of all of this; what value is `make clean` intended to add?

Reply:
> Is this intended to cover all of the places that a build creates files?

Yes, ideally.

> A `git clean -fdX` takes care of all of this; what value is `make clean` intended to add?

The whole purpose of a build system (`make` or otherwise) is to not have to remember a bunch of different switches and flags and the order in which to call them for day-to-day tasks.

Reply:
In that case, all I'm trying to say is that I think you're missing a bunch of paths. Here's the output of `git clean -ndX` on a recent build for me:

```
Would remove .psci_modules/
Would remove .stack-work/
Would remove .test_modules/
Would remove lib/purescript-ast/.stack-work/
Would remove lib/purescript-ast/purescript-ast.cabal
Would remove lib/purescript-cst/.stack-work/
Would remove lib/purescript-cst/purescript-cst.cabal
Would remove purescript.cabal
Would remove tests/purs/docs/output/
Would remove tests/purs/make/
Would remove tests/purs/publish/basic-example/output/
Would remove tests/support/bower_components/
Would remove tests/support/node_modules/
Would remove tests/support/package-lock.json
Would remove tests/support/pscide/output/
```

I haven't done a checkout of this PR to test, though, so maybe some/all of these have been moved into `.build`? I don't know. Just trying to make sure `make clean` works as intended.

Reply:
Right, it sounds like adding a `clean` target in the Makefile whose implementation is `git clean -fdX` is probably the way to go here.

Reply:
Sorry, I wasn't intending to have this rule fix all of the problems with the current build system. I only meant to clean up the new artifacts introduced with this PR.

Given that this is generating enough discussion by itself, and the build system hasn't been cleaning up after itself up until this point, I'm going to split this out to its own PR so we can focus on getting the `hlint` changes reviewed.